recipe-client-addon/lib/ActionSandboxManager.jsm

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use strict";
+
+const {utils: Cu} = Components;
+Cu.import("resource://shield-recipe-client/lib/NormandyDriver.jsm");
+Cu.import("resource://shield-recipe-client/lib/SandboxManager.jsm");
+Cu.import("resource://shield-recipe-client/lib/LogManager.jsm");
+
+this.EXPORTED_SYMBOLS = ["ActionSandboxManager"];
+
+const log = LogManager.getLogger("recipe-sandbox-manager");
+
+/**
+ * An extension to SandboxManager that prepares a sandbox for executing
+ * Normandy actions.
+ *
+ * Actions register a set of named callbacks, which this class makes available
+ * for execution. This allows a single action script to define multiple,
+ * independent steps that execute in isolated sandboxes.
+ *
+ * Callbacks are assumed to be async and must return Promises.
+ */
+this.ActionSandboxManager = class extends SandboxManager {
+  constructor() {
+    super();
+
+    // Prepare the sandbox environment
+    const driver = new NormandyDriver(this);
+    this.cloneIntoGlobal("sandboxedDriver", driver, {cloneFunctions: true});
+    this.evalInSandbox(`
+      // Shim old API for registering actions
+      function registerAction(name, Action) {
+        this.ActionClass = Action;
```

Comment:
I don't see this (`ActionClass`) being used, and you removed the tests for `registerAction`, it seems? We should probably have tests that the shim works correctly.